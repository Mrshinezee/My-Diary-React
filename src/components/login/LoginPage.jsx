// react libraries
import React, { Fragment } from 'react';

// third-party libraries
import PropTypes from 'prop-types';

// components
import Logo from '../reusables/Logo';
import LoginForm from './LoginForm';

/**
 * @param {object} props
 * @desc renders login page
 * @return {object} login
 */
const LoginPage = props => (
    <Fragment>

        <div className="menu-links">
            <div id="menu-links-left">
                <span><a href="index.html">MY-DIARY</a></span>
            </div>

            <div className="menu-icon">
                <label htmlFor="menu-check">
                    <span></span>
                    <span></span>
                    <span></span>
                </label>
            </div>

            <input type="checkbox" id="menu-check" />
            <div className="menu-links-right">
                <a href="sign-up.html">Sign Up</a>
                <a href="login.html">Login</a>
            </div>
        </div>
        <div id="main">
            <div className="centring-div">
                <div id="topic">
                    <div id="logo"></div>
                </div>
            </div>

            <div className="centring-div">
                <h2 id="motto">Your Daily Companion</h2>
            </div>
            <LoginForm
                login={props.login}
                auth={props.auth}
            />
            <div className="centring-div">
                <span id="no_account">Not registered? </span>
                <a href="sign-up.html"> Sign up today?</a>
            </div>
        </div>
    </Fragment>
);

LoginPage.propTypes = {
    login: PropTypes.func.isRequired,
    auth: PropTypes.object
};

export default LoginPage;
