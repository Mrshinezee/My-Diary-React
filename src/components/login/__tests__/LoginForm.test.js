import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import LoginPage from '../LoginPage';

let wrapper;
configure({ adapter: new Adapter() });
describe('Login component', () => {

  beforeEach(() => {
    const props = {
      login: jest.fn(),
      errors: {},
      auth: {
        isAuthenticated: true
      },
      history: {
        push: jest.fn()
      }
    };
    wrapper = shallow(
        <LoginPage {...props} />
    );
  });

  test("renders the Login component", () => { 
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
})
