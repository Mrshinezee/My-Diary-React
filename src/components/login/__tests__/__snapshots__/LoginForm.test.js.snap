// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders the Login component 1`] = `
<Fragment>
  <div
    className="menu-links"
  >
    <div
      id="menu-links-left"
    >
      <span>
        <a
          href="index.html"
        >
          MY-DIARY
        </a>
      </span>
    </div>
    <div
      className="menu-icon"
    >
      <label
        htmlFor="menu-check"
      >
        <span />
        <span />
        <span />
      </label>
    </div>
    <input
      id="menu-check"
      type="checkbox"
    />
    <div
      className="menu-links-right"
    >
      <a
        href="sign-up.html"
      >
        Sign Up
      </a>
      <a
        href="login.html"
      >
        Login
      </a>
    </div>
  </div>
  <div
    id="main"
  >
    <div
      className="centring-div"
    >
      <div
        id="topic"
      >
        <div
          id="logo"
        />
      </div>
    </div>
    <div
      className="centring-div"
    >
      <h2
        id="motto"
      >
        Your Daily Companion
      </h2>
    </div>
    <LoginForm
      auth={
        Object {
          "isAuthenticated": true,
        }
      }
      login={[MockFunction]}
    />
    <div
      className="centring-div"
    >
      <span
        id="no_account"
      >
        Not registered? 
      </span>
      <a
        href="sign-up.html"
      >
         Sign up today?
      </a>
    </div>
  </div>
</Fragment>
`;
